packages/ui/src/components/Modal/components/Dialog.tsx

```diff
@@ -209,38 +240,45 @@ export const Dialog = ({
     event.stopPropagation()
   }
 
-  return open
-    ? createPortal(
-        <StyledBackdrop
-          data-open={open}
-          onClick={handleClose}
-          className={backdropClassName}
-          css={backdropCss}
-          data-testid={dataTestId ? `${dataTestId}-backdrop` : undefined}
-          onFocus={stopFocus}
-        >
-          <StyledDialog
-            css={dialogCss}
-            onKeyUp={handleKeyUp}
-            onKeyDown={handleFocusTrap}
-            className={className}
-            id={id}
-            data-testid={dataTestId}
-            aria-label={ariaLabel}
-            data-placement={placement}
-            data-size={size}
-            open={open}
-            onClick={stopClick}
-            onCancel={stopCancel}
-            onClose={stopCancel}
-            aria-modal
-            ref={dialogRef}
-            tabIndex={0}
-          >
-            {open ? children : null}
-          </StyledDialog>
-        </StyledBackdrop>,
-        containerRef.current,
-      )
-    : null
+  const position = [...openedModales].reverse().indexOf(id) // reverse method mutate array so we need to create a new array
```

Comment:
We could use `lastIndexOf` instead of reversing no ?

Otherwhile just loop on every modals is not that costly, why going from the last one ?

Reply:
Oh ok you store the last modal as last item but you need to considerer it as position : 0, hmm okay why not then, you can probably ignore last feedback, may be add it could worth to add a comment to explain this part

Reply:
yes that's a bit tricky, the last modal open is first in position as it's size won't change and the first opened one is reversed. I will add a comment to explain better